src/fides/api/api/v1/endpoints/dataset_config_endpoints.py

```diff
@@ -108,31 +88,17 @@ def _get_connection_config(
     return connection_config
 
 
-def validate_data_categories(dataset: Dataset, db: Session) -> None:
```

Comment:
I moved a lot of the scattered validations to under `service/dataset/validation_steps`